### DISCUSSION TITLE
Slow render times / performance tips?

### DISCUSSION THREAD

**Discussion by iway1** [lottie-react-native/lottie-react-native#984] [Q&A]
I have some lottie animations who's initial render is slow and lags the UI of the application for about half a second. I'm not sure what the root cause is.

Does anyone know what the most likely performance issues are in Lottie? File size? Number of distinct elements?

**Comment by TheRogue76** [comment]
Hi @iway1,

If your Lottie file is a large JSON, chances are that is the prevalent issue. By default, metro loads certain file types on the JS side, including JSON files, and it needs to be sent over the bridge for us to be able to use it, and that causes an unneeded overhead and slows things down.

While we are waiting for the new architecture to become more mainstream so we can see if that tech can help us alleviate the problem, we have introduced dotLottie support in the meantime. dotLottie is a newer, smaller format type, and since it is not one of the file formats that Metro auto-loads, we have set it up in a way for the asset to be loaded on the native side, which should alleviate any sort of performance penalties. You can read more about it [here](https://lottiefiles.com/blog/updates/introducing-dotlottie-open-source-file-format) and [here](https://github.com/lottie-react-native/lottie-react-native?tab=readme-ov-file#if-you-want-to-use-lottie-files).

If you already have JSON files, you can also switch them to dotLottie quite easily with tools such as [this](https://lottiefiles.com/tools/lottie-to-dotlottie)
